src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -166,6 +166,9 @@
   <Target Name="SonarQubeCategoriseProject"
           Condition=" $(SonarQubeTempPath) != '' ">
 
+    <Message Text="Sonar: ($(MSBuildProjectFile)) Categorizing project as test or product code..." />
+    <Message Condition="$(SonarQubeTestProject) != ''" Importance="normal" Text="Sonar: ($(MSBuildProjectFile)) SonarQubeTestProject has been set explicitly to $(SonarQubeTestProject)" />
```

Comment:
We've had a number of support/community questions recently about why a project is being treated as test code.
The targets now write out a message for each case to make it clear.